File: dropsource_storybook/lib/src/story_app/logical_resolution.dart
import 'channel_methods.dart';

/// Flutter uses logical pixels. Also called device-independent or
/// resolution-independent pixels.
///
/// By definition, there are roughly 38 logical pixels per centimeter,
/// or about 96 logical pixels per inch, of the physical display.
///
/// This class represents the logical resolution in terms of logical 
/// [width] and logical [height].
class LogicalResolution implements OutboundChannelArgument {
  final double width;
  final double height;

  const LogicalResolution({this.width, this.height});

  @override
  Map<String, double> toStandardMap() {
    return {
      'width': width,
      'height': height
    };
  }
}

File: platform/windows/src/logical_resolution.cpp
#include "logical_resolution.h"

LogicalResolution::LogicalResolution()
{
	width = 0;
	height = 0;
}

LogicalResolution::LogicalResolution(double w, double h)
{
	width = w;
	height = h;
}

LogicalResolution::LogicalResolution(flutter::EncodableMap args)
{
	width = std::get<double>(MapUtils::getValue(args, "width"));
	height = std::get<double>(MapUtils::getValue(args, "height"));
}
